[17.10] add 17.10 changelog

vieux opened the pull request:
Signed-off-by: Victor Vieux <victorvieux@gmail.com>

Please do not send pull requests to this docker/docker-ce repository.

We do, however, take contributions gladly.

See https://github.com/docker/docker-ce/blob/master/CONTRIBUTING.md

Thanks!

vieux commented:
ping @thaJeztah @fcrisciani @tonistiigi

thaJeztah commented on CHANGELOG:
```diff
@@ -0,0 +1,51 @@
+
+### Logging
+
+* Move pkg/jsonlog to be a subpackage of the single consumer [moby/moby#34946](https://github.com/moby/moby/pull/34946)
```
This feels like an internal change, not sure it's worth mentioning

thaJeztah commented on CHANGELOG:
```diff
@@ -0,0 +1,51 @@
+### Logging
+
+* Move pkg/jsonlog to be a subpackage of the single consumer [moby/moby#34946](https://github.com/moby/moby/pull/34946)
+- Fixed `raw` mode splunk logger [moby/moby#34520](https://github.com/moby/moby/pull/34520)
```
This was back ported to 17.09 through https://github.com/docker/docker-ce/pull/233, so not new in 17.10

thaJeztah commented on CHANGELOG:
```diff
@@ -0,0 +1,51 @@
+
+### Networking
+
+* Ensure Host Network Service exists [moby/moby#34928](https://github.com/moby/moby/pull/34928)
```
This is a bug fix (Windows only); https://github.com/moby/moby/issues/34920

thaJeztah commented on CHANGELOG:
```diff
@@ -0,0 +1,51 @@
+### Runtime
+
+* LCOW: Add UVM debugability by grabbing logs before tear-down [moby/moby#34846](https://github.com/moby/moby/pull/34846)
+* LCOW: Fix comment in graphdriver code [moby/moby#34891](https://github.com/moby/moby/pull/34891)
```
This is a comment-only change in the source; we can drop this one

thaJeztah commented on CHANGELOG:
```diff
@@ -0,0 +1,51 @@
+* LCOW: Fix comment in graphdriver code [moby/moby#34891](https://github.com/moby/moby/pull/34891)
+* LCOW: Prepare work for bind mounts [moby/moby#34258](https://github.com/moby/moby/pull/34258)
+* LCOW: Support for docker cp, ADD/COPY on build [moby/moby#34252](https://github.com/moby/moby/pull/34252)
+* LCOW: VHDX boot to readonly. [moby/moby#34754](https://github.com/moby/moby/pull/34754)
```
Perhaps we should combine all the LCOW lines to something like: "Add initial support for Linux Containers On Windows (LCOW)" - or better wording

@friism ^^

thaJeztah commented on CHANGELOG:
```diff
@@ -0,0 +1,51 @@
+* LCOW: Support for docker cp, ADD/COPY on build [moby/moby#34252](https://github.com/moby/moby/pull/34252)
+* LCOW: VHDX boot to readonly. [moby/moby#34754](https://github.com/moby/moby/pull/34754)
+- Fix volume inspect with empty ID [moby/moby#34770](https://github.com/moby/moby/pull/34770)
+- Fix variable shadowing causing LayersSize to be reported as 0 [moby/moby#34826](https://github.com/moby/moby/pull/34826)
```
This was back ported to 17.09 through https://github.com/docker/docker-ce/pull/232, so not a new change

thaJeztah commented on CHANGELOG:
```diff
@@ -0,0 +1,51 @@
+* LCOW: Prepare work for bind mounts [moby/moby#34258](https://github.com/moby/moby/pull/34258)
+* LCOW: Support for docker cp, ADD/COPY on build [moby/moby#34252](https://github.com/moby/moby/pull/34252)
+* LCOW: VHDX boot to readonly. [moby/moby#34754](https://github.com/moby/moby/pull/34754)
+- Fix volume inspect with empty ID [moby/moby#34770](https://github.com/moby/moby/pull/34770)
```
Looks like this was cherry-picked into 17.09 https://github.com/docker/docker-ce/commit/7034626ac60e8da8a15e874662e0fb9635a7fefc#diff-65244c7ec61daa507c915173882e5de3R215 (only in components/cli, but it looks like a client fix)

/cc @dnephin

dnephin replied:
Yes, I agree this fix is already in 17.09.

thaJeztah commented on CHANGELOG:
```diff
@@ -0,0 +1,51 @@
+* LCOW: VHDX boot to readonly. [moby/moby#34754](https://github.com/moby/moby/pull/34754)
+- Fix volume inspect with empty ID [moby/moby#34770](https://github.com/moby/moby/pull/34770)
+- Fix variable shadowing causing LayersSize to be reported as 0 [moby/moby#34826](https://github.com/moby/moby/pull/34826)
+* Volume: evaluate symlinks before relabeling mount source [moby/moby#34792](https://github.com/moby/moby/pull/34792)
```
Should we mention `SELinux` here?

thaJeztah commented on CHANGELOG:
```diff
@@ -0,0 +1,51 @@
+- Fix volume inspect with empty ID [moby/moby#34770](https://github.com/moby/moby/pull/34770)
+- Fix variable shadowing causing LayersSize to be reported as 0 [moby/moby#34826](https://github.com/moby/moby/pull/34826)
+* Volume: evaluate symlinks before relabeling mount source [moby/moby#34792](https://github.com/moby/moby/pull/34792)
+* Set selinux label on local volumes from mounts API [moby/moby#34684](https://github.com/moby/moby/pull/34684)
```
This was back ported to 17.09 through https://github.com/docker/docker-ce/pull/241, so not new

thaJeztah commented on CHANGELOG:
```diff
@@ -0,0 +1,51 @@
+- Fix variable shadowing causing LayersSize to be reported as 0 [moby/moby#34826](https://github.com/moby/moby/pull/34826)
+* Volume: evaluate symlinks before relabeling mount source [moby/moby#34792](https://github.com/moby/moby/pull/34792)
+* Set selinux label on local volumes from mounts API [moby/moby#34684](https://github.com/moby/moby/pull/34684)
+* Close pipe in overlay2 graphdriver [moby/moby#34863](https://github.com/moby/moby/pull/34863)
```
This was back ported to 17.09 through https://github.com/docker/docker-ce/pull/245, so not new

thaJeztah commented on CHANGELOG:
```diff
@@ -0,0 +1,51 @@
+
+### Swarm Mode
+
+* Use limit with `docker swarm init --force-new-cluster` [moby/moby#34881](https://github.com/moby/moby/pull/34881)
```
Perhaps

```Markdown
* Produce an error if `docker swarm init --force-new-cluster` is executed on worker nodes [moby/moby#34881](https://github.com/moby/moby/pull/34881)
```

thaJeztah commented:
https://github.com/docker/cli/pull/565 was just merged, and needs an entry if we didn't cut off the 17.10 branch yet;

```Markdown
* Change the default output format for `docker container stats` to show `CONTAINER ID` and `NAME` [docker/cli#565](https://github.com/docker/cli/pull/565)
+ Add `--no-trunc` flag to `docker container stats` [docker/cli#565](https://github.com/docker/cli/pull/565)
```

dnephin commented:
`docker trust` is in experimental for 17.10 , right? Should we mention it?

thaJeztah commented:
Ah, yes, that should definitely be added, thanks!

vieux commented:
@thaJeztah I addressed most of your comments, feel free to edit the rest directly if you want, I wasn't sure what to do, otherwise, I think it's good for RC1

vieux commented:
LGTM

tonistiigi commented:
LGTM

andrewhsu commented:
cc @JimGalasyn

mstanleyjones commented:
@andrewhsu FYI CE stuff is me and @gbarr01 :)
